workflow/controller/controller.go

```diff
@@ -354,6 +357,9 @@ func (wfc *WorkflowController) Run(ctx context.Context, wfWorkers, workflowTTLWo
 	for i := 0; i < wfWorkers; i++ {
 		go wait.Until(wfc.runWorker, time.Second, ctx.Done())
 	}
+	for i := 0; i < archivedWorkflowWorkers; i++ {
```

Comment:
Overall I would prefer it if this was separated out into it's own controller, modelled on the [garbage collector](https://github.com/argoproj/argo-workflows/blob/2d7e2b55914a89653613f75d2d8b19ccb9c01c39/workflow/controller/controller.go#L347). I know podGC is still wrapped up into this controller, but that too feels like it could be separated.

They could still share an informer, but the concerns would be better separated.

I realise this isn't all the fault of this specific PR, but I'd like it to be done if possible.

Reply:
agree, separated controllers/files are a lot easier to read. some of them being here and others being separated is confusing

Reply:
I agree to submit a separate refactor PR, since there is a lot to change in workflow controller, and that is not specific to this PR

Reply:
Thank you

Reply:
For reference, see #13536 for this separated change